src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -49,12 +48,14 @@ internal fun executeActor(
         val res = m.invoke(instance, *args.toTypedArray())
         return if (m.returnType.isAssignableFrom(Void.TYPE)) VoidResult else createLincheckResult(res)
     } catch (invE: Throwable) {
-        val eClass = (invE.cause ?: invE).javaClass.normalize()
-        for (ec in actor.handledExceptions) {
-            if (ec.isAssignableFrom(eClass))
-                return ExceptionResult.create(eClass)
-        }
-        throw IllegalStateException("Invalid exception as a result of $actor", invE)
+        /*
+        The actor Method is invoked using Reflection API,
+        so if some exception occurs, it will be wrapped with InvocationTargetException,
+        using its `cause` property.
+        But exception may be thrown not only by method invocation,
+        so in that case we have to store the exception itself.
```

Comment:
If the exception is thrown not during the method invocation, shouldn't we fail immediately?